scripts/date-snapshot/date-snapshot.sh

```diff
@@ -1,44 +1,74 @@
 #!/bin/bash
-
+#
+# SPDX-License-Identifier: LGPL-2.1
+# Copyright (C) 2022, VMware Inc, June Knauth <june.knauth@gmail.com>
+#
 # Script to download one or several git repos at the last commit before a specified date
 # Takes inputs specified via command line or in a file
-# Created by June Knauth (VMware) <june.knauth@gmail.com>, 2022-06-22
 
 package=date-snapshot
+clean=false # clean and download define the script's behavior
+download=true
+# If clean is set, we delete existing repos. If not, we leave them untouched (no checkout).
+# If download is set, we download and checkout the repos specified if they don't exist.
 
 download_checkout(){
   IN="${1}"
-  # $fields=(${IN//;/ }) 
   IFS=';' read -ra ADDR <<< "$IN"
-  
-  git clone -b ${ADDR[2]} ${ADDR[1]} ${ADDR[0]}
-  cd ${ADDR[0]}
 
-  last_date=`date -d ${ADDR[3]} +%s` # convert given date to unix timestamp
-  # we could simply give the param to git but it understands fewer formats than date
- 
-  # this checks out the first entry in the list of commit hashes which occur before our date 
-  git checkout `git log --date=unix --before=${last_date} --pretty=format:"%H" | head -n 1`
-   
-  cd .. 
+  if $clean ; then # delete the repos if requested
+    echo "Removing ${ADDR[0]}"
+    rm -rf ./${ADDR[0]}
```

Comment:
This logic assumes that the script is executed within the `scripts/date-snapshot` directory. That is not the case when the script is started from the top Makefile, then it is executed from the project's top directory and the logic fails.  You can get the path from `${0}` of the script and append it here, instead of that `./`. Or even better - look what is passed as `-f` argument and use that path, the directories should be there.